### DISCUSSION TITLE
Best way to pull one random record from Deta Base? (Python)

### DISCUSSION THREAD

**Discussion by eddy-22** [deta/cloud-docs#336] [Help] [answered]
Hi all,

I have a LARGE Base and was wondering what the best way to get one single record, at random, would be. Something similar to MongoDB's aggregation:
```mongodb
db.collection.aggregate( [  { "$sample": {
  "size": 1
} } ] )
```
Or something similar to pymongo's `count_documents` and `find_one` functions:
```py
count = collection.count_documents( {} )   # using an empty filter
item = collection.find_one( {} , skip=count) 
```
---
I have thought of two ways to do this, which would be the best route, or is there another way that would be more performant? My thoughts were:

1. Fetching all the items using the example in the documentation, then taking a random record from that list:
```py
res = db.fetch()
all_items = res.items

while res.last:
  res = db.fetch(last=res.last)
  all_items += res.items

import random
desired_item = random.choice(all_items)
```
However this is going to be very time consuming and inefficient due to my base's size.

2. Use `db.get(key)` with a range of incremented keys, and taking a random key from that to pass to `db.get`. I'm not sure how to implement this at the moment.

--- 
If anyone has suggestions or better ideas please let me know, thanks!

**Comment by jnsougata** [comment]
First I want to know what type of keys you are using to store the records?
Is those maintaining a sequence or any data field inside your base is having a data maintaining a sequence for all the keys ?

**Reply by eddy-22** [reply]
Yes, so I actually have a timestamp field that is sequenced ascending. The key fields are currently strings of n, n+1, n+2 etc. The first record key="1", the next="2", the next="3" and so on (as strings).

**Reply by jnsougata** [reply]
So if you know the range of the key then you can just generate a random integer ` randint()` then convert it to `str` and use `.get()`  to fetch the record from base (assuming the range will be fixed otherwise you have to check and retry)

```python
key = str(random.randint(first, last+1))
db.get(key)
```

**Comment by abdelhai** [comment, marked as answer]
deta base doesn't support this capability. the randomization should happen outside the query. in addition to what @jnsougata suggested, it might also be possible to generate a specific time in the past and use comparison operators to get a random result. not sure how well this would work in reality.
